src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java

```diff
@@ -52,7 +53,7 @@
  * {@link PropertyCondition#register(Class, String, String)}, be aware that there can only be two patterns -
  * the first one needs to be a non-negated one and a negated one.
  */
-public abstract class PropertyCondition<T> extends Condition implements Checker<T> {
+public abstract class PropertyCondition<T> extends Condition implements ch.njol.util.Checker<T>, Predicate<T> {
```

Comment:
The implementation of `Predicate` here raises some fun questions. It contains several default methods that allow you to modify the predicate. I'm not sure how applicable they are, but we might wish to override them or disallow them completely.

For example, should `negate()` function the same as `setNegated(!isNegated())`? It might be the case that `negate()` expects a new object rather than the same one (which we can't really promise)

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Predicate.html

Reply:
No walrus they don't modify the predicate they retain adjusted wrappers for it, e.g. the negation of `o -> true` would be `(!(o -> true))`.

Reply:
I mean that the methods such as `negate` are a bit unexpected on PropertyCondtion, especially give we already have setNegated